### PR TITLE
describe where certs come from

### PR DESCRIPTION
**Pull request by deads2k**
```
[deads@deads-02 cluster-kube-apiserver-operator]$ make && ./cluster-kube-apiserver-operator where-do-resources-come-from
go build  ./cmd/cluster-kube-apiserver-operator
configmaps/kube-apiserver-aggregator-client-ca[openshift-config-managed] - synchronized
    configmaps/aggregator-client-ca[openshift-kube-apiserver] - unioned
        configmaps/initial-aggregator-client-ca[openshift-config] - static
            Installer/cluster[]
        configmaps/managed-aggregator-client-ca[openshift-kube-apiserver-operator] - rotated
            deployments.apps/openshift-kube-apiserver-operator[openshift-kube-apiserver-operator] - AggregatorClientRotation
                deployments.apps/cluster-version[openshift-cluster-version] - controller.CVO
                    Payload/cluster[]

```

@sttts 

rought, but fairly effective I think

`cluster-kube-apiserver-operator resource-graph | dot -Kfdp -T svg -o ../resources.svg`

resources

### CODE REVIEW

File: cmd/cluster-kube-apiserver-operator/main.go [hunk 1]
```diff
@@ -7,6 +7,8 @@ import (
 	"os"
 	"time"
 
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/cmd/wheredoresourcescomefrom"
```
Comment (sttts) [inline cmd/cluster-kube-apiserver-operator/main.go]: order

File: pkg/operator/reactionchain/reactionchain.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+	ret.SetNested()
+	return ret
+}
+
+type resourcesImpl struct {
+	resources []Resource
+}
+
+func (r *resourcesImpl) SetNested() {
```
Comment (sttts) [inline pkg/operator/reactionchain/reactionchain.go]: what is this?

File: pkg/cmd/wheredoresourcescomefrom/areyoumymother.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	"fmt"
+	"strings"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/reactionchain"
+	"github.com/spf13/cobra"
+)
+
+func NewResourceChainCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "where-do-resources-come-from",
```
Comment (sttts) [inline pkg/cmd/wheredoresourcescomefrom/areyoumymother.go]: not sure I understand where this is going. I expected just some simple data structure with namespaces and names that we can refer to from the controllers.

This code here does not really serve the purpose of documentation.

File: pkg/operator/reactionchain/reactionchain.go (hunk 2 above)
Comment (deads2k) [reply pkg/operator/reactionchain/reactionchain.go]: > what is this?

link inverter

File: pkg/cmd/wheredoresourcescomefrom/areyoumymother.go (hunk 3 above)
Comment (deads2k) [reply pkg/cmd/wheredoresourcescomefrom/areyoumymother.go]: > not sure I understand where this is going. I expected just some simple data structure with namespaces and names that we can refer to from the controllers.
> 
> This code here does not really serve the purpose of documentation.

Payload leads to CVO leads to aggregator cert rotation controller leads to configmaps/managed-aggregator-client-ca, separately the installer leads to configmaps/initial-aggregator-client-ca, configmaps/initial-aggregator-client-ca and configmaps/managed-aggregator-client-ca are combined in configmaps/aggregator-client-ca which is then copied to configmaps/kube-apiserver-aggregator-client-ca

Pretty sure this is the smallest way to express that.

File: pkg/cmd/wheredoresourcescomefrom/areyoumymother.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+		Short: "ask your mother",
```
Comment (soltysh) [inline pkg/cmd/wheredoresourcescomefrom/areyoumymother.go]: lol

Comment (deads2k) [reply pkg/cmd/wheredoresourcescomefrom/areyoumymother.go]: @soltysh I wrote this entire pull just to write that sentence :)

Comment (mfojtik) [reply pkg/cmd/wheredoresourcescomefrom/areyoumymother.go]: `wheredoresourcescomefrom` -> `ask your mother`

**Comment by deads2k** [comment]
```
configmaps/kube-apiserver-aggregator-client-ca[openshift-config-managed] - synchronized
    configmaps/aggregator-client-ca[openshift-kube-apiserver] - unioned
        configmaps/initial-aggregator-client-ca[openshift-config] - static
            Installer/cluster[]
        configmaps/managed-aggregator-client-ca[openshift-kube-apiserver-operator] - rotated
            deployments.apps/openshift-kube-apiserver-operator[openshift-kube-apiserver-operator] - AggregatorClientRotation
                deployments.apps/cluster-version[openshift-cluster-version] - controller.CVO
                    Payload/cluster[]
```

@sttts I tried to crisp up the expression and here is sample output. It's a little hard to visualize this graph.  We may do well to have DOT output.  See if the different expression of dependencies helps.

**Comment by deads2k** [comment]
network flake

/retest

**Pull request merged**